src/com/amazon/ion/util/Equivalence.java

```diff
@@ -431,8 +515,14 @@ private static int ionCompareToImpl(final IonValue v1,
                              ((IonInt) v2).bigIntegerValue());
                     break;
                 case FLOAT:
-                    result = Double.compare(((IonFloat) v1).doubleValue(),
-                                            ((IonFloat) v2).doubleValue());
+                    double double1 = ((IonFloat) v1).doubleValue();
+                    double double2 = ((IonFloat) v2).doubleValue();
+                    if (configuration.epsilon != null
+                            && (double1 == double2 || Math.abs(double1 - double2) < configuration.epsilon)) {
```

Comment:
Should `epsilon` be an inclusive maximum? (i.e. `<= configuration.epsilon`)

Reply:
Yes it should. Will fix before merge.